community-containers/dlna/readme.md

```diff
@@ -1,5 +1,11 @@
 ## DLNA server
-This container bundles DLNA server for your Nextcloud files to be accessible by the clients in your local network. Simply run the container and look for a new media server Nextcloud-AIO in your local network.
+This container bundles DLNA server for your Nextcloud files to be accessible by the clients in your local network. Simply run the container and look for a new media server `nextcloud-aio` in your local network.
+
+### Notes
+- This container needs the `nextcloud-aio-nextcloud` container to be up and running to work
+- This is not working on Docker Desktop since it needs `network_mode: host` in order to work correctly
+- This is not working on Mac OS since the `host` networking mode doesn't really share the host's network interfaces in this system
+- It is needed to allow TCP network traffic on port 9999 to the AIO instance and the UDP (port 1900) from it to the local network
```

Comment:
BTW, now that I am reading this, don't you think using bridged networking and exposing 9999/tcp and 1900/udp should make it work?

Reply:
Not sure, I'm not an expert in docker network settings. I've just glanced at plex and miniDLNA docker containers as an inspiration ;)
From what I know the requirements are:
* container must be able to make multicast requests to the local network on UDP, port 1900
* container must be able to receive the UDP requests from other machines in the local network
* in the above messages, DLNA messages contain the `PresentationURL` which should be a local network accessible address for TCP (HTTP) requests such as 192.168.0.10:9999.

Reply:
Have you already tried using
`"internal_port": "9999",`
and then adding 
```
            "ports": [
                {
                  "ip_binding": "",
                  "port_number": "9999",
                  "protocol": "tcp"
                },
                {
                    "ip_binding": "",
                    "port_number": "1900",
                    "protocol": "udp"
                }
            ],
```

Reply:
I will check this out once I start my Linux box again :)

Reply:
OK, I've  tested the `bridge` net mode and standard container net mode and they both failed. Even if the bridged network seems to use the same network interface as the host, since the container has a different address, the `PresentationURL` is unusable for other devices in the local network to reach the DLNA server with it. The container mode not only has a different address issue (just like `bridge`) but also it doesn't share the network interface so the container is unable to listen for responses from the local network devices at all. So, I guess the `host` option is the only way to go. Maybe it would be possible to create a network that would be accessible for both containers and other local network devices, but it doesn't seem to be simple (at least for me) and I suppose It would be easy to implement it in the AIO as a generic solution for every network configuration. But, as I said, I'm not an expert and maybe there is some better solution - I just don't know it.

Reply:
> . So, I guess the `host` option is the only way to go.

All right. Thanks for testing!